src/Skeleton/Skeleton.component.tsx

```diff
@@ -134,17 +136,16 @@ export const Skeleton = ({
 
     const length = textLength || (childText && childText.length);
     const blockText =
-        length &&
-        Array(Math.ceil(length / blockToAverageCharQuotient) + 1).join(
-            blockChar
-        );
+        length && Array(Math.ceil(length * blockRatio) + 1).join(textChar);
+    const textOnly = !!blockText;
 
```

Comment:
I think this might be over engineering so it take it with a grain of salt (or just a point of discussion); there are a few calculations that always happen when render is triggered. Maybe worth memoizing them? I don't want to flag every spot, but there's a `replace()` with a regex, some array `join()`'s with `Math`, things of that nature. It may not make sense in some cases but I know we're all thinking more about performance so I'm throwing it out there.

Reply:
I was going to do this but then I decided I want to early return before anything else (so that the normal case of just passing through is as inexpensive as possible). This precludes using a hook like `useMemo`